app-frontend/src/app/components/map/annotatePopup/annotatePopup.controller.js

```diff
@@ -0,0 +1,154 @@
+/* globals jQuery */
+
+const _ = require('lodash');
+
+export default class AnnotatePopupController {
+    constructor(
+        $log, $scope, $compile,
+        mapService
+    ) {
+        'ngInject';
+        this.$log = $log;
+        this.$scope = $scope;
+        this.$compile = $compile;
+        this.getMap = () => mapService.getMap(this.mapId);
+    }
+
+    $onInit() {
+        this.labelObj = {'name': this.initialLabelName};
+        this.description = this.initialDescription;
+    }
+
+    cancelAnnotation() {
```

Comment:
Like I said above, let's keep this component as thin as possible, with the map interaction happening in the parent controller, and this component simply executing callbacks using function bindings